core/modules/widgets/range.js

```diff
@@ -52,8 +52,10 @@ RangeWidget.prototype.render = function(parent,nextSibling) {
 	this.inputDomNode.value = this.getValue();
 	// Add a click event handler
 	$tw.utils.addEventListeners(this.inputDomNode,[
-		{name: "input", handlerObject: this, handlerMethod: "handleInputEvent"},
-		{name: "change", handlerObject: this, handlerMethod: "handleInputEvent"}		
+		{name:"mousedown", handlerObject:this, handlerMethod:"handleMouseDownEvent"},
+		{name:"mouseup",   handlerObject:this, handlerMethod:"handleMouseUpEvent"},
+		{name:"change",    handlerObject:this, handlerMethod:"handleChangeEvent"},
+		{name:"input",     handlerObject:this, handlerMethod:"handleInputEvent"},
```

Comment:
I'd rather see this list built dynamically based on whether there's an attribute specified for that event. Something like this:

```javascript
var eventList = [];
if(this.actionsMouseDown) {
    eventList.push({name:"mousedown", handlerObject:this, handlerMethod:"handleMouseDownEvent"});
}
// Similar for this.actionsMouseUp and this.actionsChange
if(this.actionsInput) {
    eventList.push({name:"input",     handlerObject:this, handlerMethod:"handleInputEvent"});
}
$tw.utils.addEventListeners(this.inputDomNode, eventList);
```

The advantage is that if the end user doesn't care about the `input` event, we don't need to call a dozen times into an event handler that will end up doing nothing. There is a performance cost to the browser firing events, even if the event handler returns quickly. If we can avoid hooking up the event at all, the browser can optimize and not fire the event because it knows nothing is going to handle the event.

Reply:
Good point @rmunn

Reply:
The problem with input and change event are, that they have to be fired, even if there is no action. Otherwise the target tiddler wouldn't be updated. It doesn't matter if there are actions defined or not. ... Input is fired on all modern browsers and change is fired in IE.

You are right for MouseDown and MouseUp. ATM they are only there for handling actions atm. ... BUT we don't know if this will change in the future. 

Events are only fired if user wants interaction. So IMO there is no real performance overhead, without user interaction. So I'd vote for simple code here.

Reply:
I forgot something. MouseUp is needed for IE to trigger the change event. So it doesn't matter if actions are defined.